01_explore_data/nbconverted/explore_cancer_gene_sets.py

```diff
@@ -0,0 +1,388 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Explore cancer gene sets
+# 
+# In `00_download_data/2_download_cancer_gene_data.ipynb`, we downloaded and integrated several different sets of cancer-associated genes. In this script, we want to explore the similarities and differences between these gene sets in more detail.
+# 
+# We see that the results of our classification experiments are somewhat dependent on which gene set or sets we decide to use, so understanding the similarities and differences between the gene sets will help us to put these in context.
+
+# In[1]:
+
+
+import sys
+from pathlib import Path
+
+import pandas as pd
+
+import mpmp.config as cfg
+import mpmp.utilities.data_utilities as du
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+# load datasets
+vogelstein_df = du.load_vogelstein()
+vogelstein_df.head()
+
+
+# In[3]:
+
+
+cosmic_df = du.load_cosmic()
+cosmic_df.head()
+
+
+# In[4]:
+
+
+# load Bailey et al. data from excel file
+# this is the same as the code in 00_download_data/2_download_cancer_gene_set.ipynb
+class_df = pd.read_excel(
+    cfg.bailey_raw_file,
+    engine='openpyxl', sheet_name='Table S1', index_col='KEY', header=3
+)
+class_df.drop(
+    class_df.columns[class_df.columns.str.contains('Unnamed')],
+    axis=1, inplace=True
+)
+class_df.rename(columns={'Tumor suppressor or oncogene prediction (by 20/20+)':
+                         'classification'},
+                inplace=True)
+
+bailey_df = (
+    class_df[((class_df.Cancer == 'PANCAN') &
+             (~class_df.classification.isna()))]
+).copy()
+
+bailey_df.head()
+
+
+# ### Overlap between COSMIC/Bailey/Vogelstein
+# 
+# Is COSMIC a strict subset of the Bailey and Vogelstein cancer driver datasets? Or are there genes in the latter two that are not in COSMIC?
+
+# In[5]:
+
+
+vogelstein_genes = set(vogelstein_df.gene.values)
+cosmic_genes = set(cosmic_df.gene.values)
+bailey_genes = set(bailey_df.Gene.values)
+
+
+# In[6]:
+
+
+from venn import venn
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+sns.set_style('white')
+
+label_map = {
+    'cosmic': cosmic_genes,
+    'bailey': bailey_genes,
+    'vogelstein': vogelstein_genes
+}
+venn(label_map)
+plt.title('Overlap between cancer gene sets', size=13)
+
+
+# ### Enrichment analysis of gene sets
```

Comment:
It seems like there is a lot of processing happening in these cells that could organizationally could be split out into a separate nb to be cleaner. But I completely understand if this is a one-off/quick analysis that you wanted to keep everything together. So completely up to you, just something that I noticed.

Also, I'm not as familiar with ontologies and the format of them, so there are many steps here that I'm not exactly sure why you're doing it. Not sure if this can easily read if you provide an example of what the ontology looks like?

Reply:
Yeah, this is definitely not as clean as it could be. I think the goal is just a quick one-off analysis, so I may leave it how it is, but I'll try to document it a bit better.

I do think it's possible to plot the ontology (or part of it) using the `goatools` package, so I'll look into that.

Reply:
I added an example of plotting part of the ontology that's relevant to the non-Vogelstein genes at the end of the script. Here's what it looks like:

![image](https://user-images.githubusercontent.com/2345877/156930577-117ee097-ae45-443b-9377-ebf190ad86f0.png)

GO terms and the relations between them generally look like this (a directed acyclic graph, or DAG). Plotting the whole thing could get super gnarly, so I had to cherry-pick a few terms that we're interested in.